test/test_transforms.py

```diff
@@ -136,42 +136,157 @@ def test_mu_law_companding(self):
+                                                    norm='ortho')
+        # check defaults
+        torch_mfcc = mfcc_transform(audio_scaled)
+        self.assertTrue(torch_mfcc.dim() == 3)
+        self.assertTrue(torch_mfcc.shape[2] == n_mfcc)
+        self.assertTrue(torch_mfcc.shape[1] == 321)
+        # check melkwargs are passed through
+        melkwargs = {'ws': 200}
+        mfcc_transform2 = torchaudio.transforms.MFCC(sr=sample_rate,
+                                                     n_mfcc=n_mfcc,
+                                                     norm='ortho',
+                                                     melkwargs=melkwargs)
+        torch_mfcc2 = mfcc_transform2(audio_scaled)
+        self.assertTrue(torch_mfcc2.shape[1] == 641)
+
+        # check norms work correctly
+        mfcc_transform_norm_none = torchaudio.transforms.MFCC(sr=sample_rate,
+                                                              n_mfcc=n_mfcc,
+                                                              norm=None)
+        torch_mfcc_norm_none = mfcc_transform_norm_none(audio_scaled)
+
+        norm_check = torch_mfcc.clone()
+        norm_check[:, :, 0] *= np.sqrt(n_mels) * 2
+        norm_check[:, :, 1:] *= np.sqrt(n_mels / 2) * 2
+
+        self.assertTrue(torch_mfcc_norm_none.allclose(norm_check))
+
+    def test_librosa_consistency(self):
+        try:
+            import librosa
+            import scipy
+        except ImportError:
+            return
+
+        input_path = os.path.join(self.test_dirpath, 'assets', 'sinewave.wav')
+        sound, sample_rate = torchaudio.load(input_path)
+        sound_librosa = sound.cpu().numpy().squeeze().T  # squeeze batch and channel first
+
+        n_fft = 400
+        hop_length = 200
+        power = 2.0
+        n_mels = 128
+        n_mfcc = 40
+        sample_rate = 16000
+
+        # test core spectrogram
+        spect_transform = torchaudio.transforms.Spectrogram(n_fft=n_fft, hop=hop_length, power=2)
+        out_librosa, _ = librosa.core.spectrum._spectrogram(y=sound_librosa,
+                                                            n_fft=n_fft,
+                                                            hop_length=hop_length,
+                                                            power=2)
+
+        out_torch = spect_transform(sound).squeeze().cpu().numpy().T
+        self.assertTrue(np.allclose(out_torch, out_librosa, atol=1e-5))
+
+        # test mel spectrogram
+        melspect_transform = torchaudio.transforms.MelSpectrogram(sr=sample_rate, window=torch.hann_window,
+                                                                  hop=hop_length, n_mels=n_mels, n_fft=n_fft)
+        librosa_mel = librosa.feature.melspectrogram(y=sound_librosa, sr=sample_rate,
+                                                     n_fft=n_fft, hop_length=hop_length, n_mels=n_mels,
+                                                     htk=True, norm=None)
```

Comment:
Is there a test comparing melspectrogram call with the default htk=False for example?

Reply:
no, there is not.  The purpose of this is not to create an exhaustive consistency test.  More so that it works similarly.

Reply:
Thank @dhpollack. I see, the reason I asked is that somehow I cannot match the default librosa's melspectrogram behaviour. Curious if you ever tried?

Reply:
You would have to implement non-htk mel-bins to be able to match Librosa's melspectrogram default behavior.
https://librosa.github.io/librosa/_modules/librosa/core/time_frequency.html#hz_to_mel
The librosa consistency test is more to show which options to set librosa at to match the current torchaudio implementation. Shouldn't be too hard to add non-htk mel frequency bins to torchaudio but in practice the difference doesn't seem to matter for speech recog tasks so I didn't work on it.